working/augmentation-libraries/feature-specification.md

```diff
@@ -738,85 +782,52 @@ It's a **compile-time** error if:
-abstract instance variable declarations.
-
-*A variable declaration implicitly has code for the synthesized getter and
-setter that access and modify the underlying backing storage. Since we don't
-allow augmentations to replace code, that implies that augmentations can't
-change variables. So we don't allow them to be augmented.*
-
-#### Abstract instance variables
-
-Dart supports `abstract` field declarations. They are syntactic sugar for
-declaring an abstract getter and, if not `final`, an abstract setter. They don't
-actually declare a variable with any backing storage.
-
-Because abstract variables are effectively abstract getter and setter
-declarations, they can be augmented and used in augmentations just like function
-declarations:
-
-*Examples:*
+If the variable is `abstract`, then the getter and setter are incomplete,
+otherwise they are complete. *For non-abstract variables, the compiler
+synthesizes a getter that accesses the backing storage and a setter that updates
+it, so these members have bodies.*
 
-```dart
-class C {
-  // Augment an abstract variable with a getter:
-  abstract final int a;
-  augment int get a => 1; // OK.
+A getter can be augmented by another getter, and likewise a setter can be
+augmented by a setter. This is true whether the getter or setter is explicitly
+declared or implicitly declared using a variable declaration.
 
-  // Augment an abstract variable with a getter and setter:
-  abstract int b;
-  augment int get b => 1; // OK.
-  augment set b(int value) {} // OK.
+*Since non-abstract variables are complete, that implies that it is an error to
+augment a non-abstract variable declaration with a complete getter, setter, or
+variable declaration. Likewise, it is an error to augment a complete getter or
+setter with a non-abstract variable declaration.*
 
-  // Augment a getter with an abstract variable:
-  int get c;
+It's a **compile-time error** if:
 
-  @someMetadata
-  augment abstract final int c; // (Not very useful, but valid.)
-}
-```
+*   The signature of the augmenting getter or setter does not [match][signature
+    matching] the signature of the augmented getter or setter.
 
-For purposes of [signature matching][], the implicit setter induced by a
-non-final abstract variable has a positional parameter named `_`. *This means
-you can augment an abstract variable with a setter that uses whatever positional
-parameter name you want:*
+*   An augmenting setter specifies any default values. *Default values are
+    defined solely by the introductory declaration.*
 
-```dart
-class C {
-  abstract int x;
-}
+*   A `const` variable declaration is augmented or augmenting.
 
-augment class C {
-  augment set x(int anyNameIWant) { // OK.
-    // ...
-  }
-}
-```
+*   A getter or setter (including one implicitly induced by a variable
+    declaration) is not complete after all augmentations are applied, unless it
+    is in a context where it can be abstract. *Every getter or setter
```

Comment:
Maybe: "unless it's an instance member, and the surrounding class is abstract, or it's a generative constructor"

An instance member of an abstract class can remain abstract, and we can give an abstract generative constructor a trivial default implementation.

I can't think of any other place where a member can validly be abstract or implicitly implemented.